pages/library/[libGameId]/index.jsx

```diff
@@ -3,11 +3,39 @@ import BackButton from "@/components/buttons/BackButton";
 import { useRouter } from "next/router";
 import useSWR from "swr";
 import Image from "next/image";
+import {
+  StyledButton,
+  StyledButtonWrapper,
+} from "@/components/buttons/DefaultButtons";
+import MenuButton from "@/components/buttons/MenuButton";
+import { useState, useEffect, useRef } from "react";
+import MenuOption from "@/components/MenuOption";
 
 const fetcher = (...args) => fetch(...args).then((response) => response.json());
-
-export default function LibraryGameDetails() {
+export default function LibraryGameDetails({
+  setGames,
+  games,
+  setMenuMode,
+  menuMode,
+}) {
   const router = useRouter();
+  const [buttonMode, setButtonMode] = useState("default");
+  const dialogRef = useRef(null);
+
+  function gameExists(newGame) {
+    return games?.some((game) => Number(game.id) === newGame[0].id);
+  }
+
+  useEffect(() => {
+    if (buttonMode === "add") {
+      setTimeout(() => {
+        dialogRef.current?.scrollIntoView({
+          behavior: "smooth",
+          block: "center",
+        });
+      }, 50);
+    }
```

Comment:
When ever you work with `setTimeout`, an `id` for this timeout gets created that we can and should use to remove the timeout after we no longer need it.

You can use the cleanup function of `useEffect` to do this and call `clearTimeout`:

```suggestion
    if (buttonMode === "add") {
      const timeOutId = setTimeout(() => {
        dialogRef.current?.scrollIntoView({
          behavior: "smooth",
          block: "center",
        });
      }, 50);
      
      return () => clearTimeout(timeOutId);
    }
```